x-pack/plugins/apm/public/components/shared/Links/url_helpers.ts

```diff
@@ -60,19 +60,34 @@ const DEFAULT_KIBANA_TIME_RANGE = {
   }
 };
 
-function getQueryWithRisonParams(location: Location, query: RisonDecoded = {}) {
+function getQueryWithRisonParams(
+  location: Location,
+  pathname: string,
+  query: RisonDecoded = {}
+) {
   // Preserve current _g and _a
   const currentQuery = toQuery(location.search);
   const decodedG = risonSafeDecode(currentQuery._g);
   const combinedG = { ...DEFAULT_KIBANA_TIME_RANGE, ...decodedG, ...query._g };
   const encodedG = rison.encode(combinedG);
   const encodedA = query._a ? rison.encode(query._a) : '';
 
-  return {
+  const nextQuery: StringMap = {
     ...query,
-    _g: encodedG,
-    _a: encodedA
+    _g: encodedG
   };
+
+  // Preserve kuery for apm links
+  const isApmLink = pathname.includes('app/apm') || pathname === '';
+  if (currentQuery.kuery && isApmLink) {
+    nextQuery.kuery = currentQuery.kuery;
+  }
+
+  if (encodedA) {
+    nextQuery._a = encodedA;
```

Comment:
Why this fork instead of leaving it in the `nextQuery` initialization?

Reply:
To avoid adding an empty `_a` to the url like here: https://github.com/elastic/kibana/pull/29430/files#diff-c895cc429fae008d977973bbe92258e8L534

Reply:
Ah, I guess if we used `undefined` instead of `''` in the ternary above, this would probably take care of itself? But this works too.